ui/src/utils/QualityControl.js

```diff
@@ -40,11 +40,45 @@ export class QualityControl {
             errorMessage: '',
         };
 
+        const audioResult = this.soundCheck();
+
         if (this.audioBuffer.duration < 2.0) {
             qualityResult.success = false;
             qualityResult.errorMessage += 'Audio recording failed: recording was too short. Try again';
+        } else if (audioResult) {
+            qualityResult.success = false;
+            qualityResult.errorMessage += audioResult;
         }
 
         return qualityResult;
     }
+
+   /**
+    * Checks if the sound is too silent
+    */
+    soundCheck() {
+        const bufferArray = this.audioBuffer.getChannelData(0);
+        const bufferSet = new Set(bufferArray);
+        const arrOfSet = [];
```

Comment:
nit: is this array of set? I feel like it would be helpful if this and arrOfBiggest were slightly more meaningfully named